.github/workflows/puppet-test.yml

```diff
@@ -0,0 +1,85 @@
+name: puppet-test
+
+# Only run installer tests, Rake spec, lint for main branch, tags, or if the branch has relevant changes
+on: 
+  push:
+    tags:
+      - '*'
+    paths:
+      - 'internal/buildscripts/packaging/installer/**'
+      - 'internal/buildscripts/packaging/tests/**'
+  pull_request:
+    branches:
+      - 'main'
+
+concurrency:
+  group: puppet-test-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  puppet-test:
+    name: puppet-test
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        PACKAGE_TYPE: [ "deb", "rpm" ]
+    env:
+      PYTHON_VERSION: 3.8.5
+      PIP_VERSION: 20.2.4
+      REQUIREMENTS_PATH: "internal/buildscripts/packaging/tests/requirements.txt"
+      RESULT_PATH: "~/testresults"
+    steps:
+      - name: Check out the codebase.
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Lint
+        run: |
+          make -C deployments/puppet lint
+
+      - name: Rake spec
+        run: |
+          make -C deployments/puppet rake-spec
+
+      - name: Setup pyenv
```

Comment:
Let's use `actions/setup-python@v2` like the installer script test.

Reply:
Changes made.